browser/ui/views/geolocation_accuracy_helper_dialog_view.h

```diff
@@ -0,0 +1,47 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_GEOLOCATION_ACCURACY_HELPER_DIALOG_VIEW_H_
+#define BRAVE_BROWSER_UI_VIEWS_GEOLOCATION_ACCURACY_HELPER_DIALOG_VIEW_H_
+
+#include "base/functional/callback_forward.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/raw_ref.h"
+#include "ui/base/metadata/metadata_header_macros.h"
+#include "ui/views/window/dialog_delegate.h"
+
+class PrefService;
+
+namespace views {
+class Checkbox;
+}  // namespace views
+
+class GeolocationAccuracyHelperDialogView : public views::DialogDelegateView {
```

Comment:
optional: From my experience, when having modal window, using [constrained_window](https://source.chromium.org/chromium/chromium/src/+/main:components/constrained_window/constrained_window_views.h;drc=5532088060e8fb483c3316175f2e12d31f09db41) could be a good choice as it provides more functionality to make it sure that the modal dialog is visible to users, such as restoring window when it's minimized.